External/Plugins/HaXeContext/Context.cs

```diff
@@ -702,9 +710,15 @@ public override void RemoveClassCompilerCache()
         #endregion
 
         #region SDK
-        private InstalledSDK GetCurrentSDK()
-        {
-            return hxsettings.InstalledSDKs?.FirstOrDefault(sdk => sdk.Path == currentSDK);
+        private InstalledSDK GetCurrentSDK() => hxsettings.InstalledSDKs?.FirstOrDefault(sdk => sdk.Path == currentSDK) ?? GetCustomSDK(currentSDK);
+
+        private InstalledSDK GetCustomSDK(string customPath)
+        {
+            InstalledSDK sdk;
+            if (customSDKCache.TryGetValue(customPath, out sdk)) return sdk;
+            sdk = createCustomSDK(customPath);
+            if (sdk != null) customSDKCache.Add(customPath, sdk);

```

Comment:
@gene-pavlovsky @SlavaRa I'm a bit worried about this static cache: shoulnd't it be cleared in some scenarios?

Reply:
I'll try to find time today and think about this improvement. Perhaps this is a very correct proposal.

Reply:
@elsassph Yes you're right. I see two potential issues here:
- Switch from Project A to Project B. Both projects use the same custom SDK which is Haxeshim, but they have different compiler versions set in `.haxerc`. In this case FD will think Project B is using whatever version Project A is using, and might e.g. use wrong completion options.
- A haxe SDK at the custom SDK path is upgraded while FD is open. I guess that will also be a problem with regular installed SDKs.
Do you see other problem scenarios?

It might be better to keep the cache in `PluginMain` instead of `Context`, and add ValidateSDK call in same cases as for installed SDKs. Might be also worth to clear the cache when switching projects, I think it's ok to not clear it, just re-validate as necessary.

Reply:
I tried to resolve these concerns in this PR:
https://github.com/fdorg/flashdevelop/pull/2206